docs/index.rst

```diff
@@ -1,14 +1,162 @@
-Welcome to Betelgeuse's documentation!
-======================================
+Betelgeuse documentation
+========================
 
-Betelgeuse reads standard Python test cases and offers tools to interact with
-Polarion. Possible interactions are:
+.. contents:: Topics
 
-* Automatic creation and update of Requirements and Test Cases from a Python
+.. _what_is_betelgeuse:
+
+What is Betelgeuse?
+```````````````````
+
+Betelgeuse is a python program that reads standard Python test cases and offers
+tools to interact with Polarion. Possible interactions are:
+
+* Automatic creation/updation of Requirements and Test Cases from a Python
   project code base.
 * Creation of Test Runs based on a jUnit XML file.
 
-Documentation contents:
+Betelgeuse uses `testimony <https://pypi.python.org/pypi/testimony>`_ and
+pylarion projects to parse test cases and communicate with polarion
+respectively.
+
+.. _prerequisites:
+
+Prerequisites
+`````````````
+Login to polarion in a browser to check if the login user has permissions to do
+the following:
+
+* add/update test case
+* add/update requirements
+* create test runs
+
+.. _quick_start:
+
+Quick Start
+```````````
+1. betelgeuse uses pylarion to interact with polarion. Install pylarion from its
+   source.
+
+   .. note::
+
+     - It may be possible that the latest version of betelgeuse may not work
+       correctly with some versions of pylarion.  In this case, please use an
+       alternate working version of pylarion.
+     - Read pylarion documentation and set up .pylarion config file as
+       required.
+
+2. Install betelguese from pypi.
+
+   .. code-block:: bash
+
+       $ pip install betelgeuse
+
+3. Alternatively you can fork betelgeuse to your github account and clone it
+   locally.  And then you can install it locally by:
+
+   .. code-block:: bash
+
+       $ cd betelgeuse
+       $ pip install -r requirements.txt
+       $ pip install -e .
+
+   .. note:: It is always recommended to use python virtual environment
+
+.. _usage_examples:
+
+Usage Examples
+``````````````
+.. note::
+
+  For easy understanding of betelgeuse, this repository is already included with
+  `data_sample` folder.  This folder contains sample tests and xml results which
+  will help in setting up and testing betelgeuse for your project.  The sample
+  commands used below also use this data.
+
+.. _help_command:
+
+help command
+++++++++++++
+
+.. code-block:: bash
+
+    $ betelgeuse --help
+
+.. _test_case_command:
+
+test-case command
++++++++++++++++++
+Creates/Syncs test cases with polarion. This command performs the following
+steps:
+
+- Testimony is called to parse the test cases.
+- For each parsed test case, the following actions are performed:
+
+    - The requirement name is derived from the test module name.  If the test
```

Comment:
It will first check if the test method have a `@Requirement` token and create/fetch the requirement after it. If that is not present then it will default to the module name.